Build DMG files with `appdmg`

fluiddot opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

Related to https://github.com/Automattic/studio/pull/135.

## Proposed Changes

- Use `appdmg` library to create dmg files.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Observe CI jobs succeed.
- Observe `Build for Mac` job generates dmg files.
- Download and run dmg files to confirm they work as expected.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

fluiddot committed:
Use `appdmg` in `made-dmg` script

fluiddot committed:
Do not install `create-dmg` in Buildkite

fluiddot commented:
I'm encountering the same issue spotted [here](https://github.com/Automattic/studio/pull/135#issuecomment-2112913759). The problem is related to a native module used in the package `fs-xattr`. Seems npm builds native modules for the current architecture only. As a workaround, we could try to enforce the needed architecture (`x86` or `arm64`) based on the matrix.

fluiddot committed:
Rebuild specific node native modules

fluiddot committed:
[TEST] Log native module rebuild

fluiddot commented on .buildkite/commands/install-node-dependencies.sh:
```diff
@@ -42,6 +42,9 @@ npm ci \
   --maxsockets "$MAX_SOCKETS" \
   "$@"
 
+# Rebuid specific native modules to avoid failures due to using different architectures
+npm rebuild fs-xattr --verbose --cpu universal
```
This workaround works locally but I couldn't make it work on CI 😞 . In theory, if we rebuild the native module using the architecture `x64` or `universal`, there shouldn't be any failures when generating the DMG file for `x64` or `universal` versions.

mokagio commented:
Just a screenshot to show the correct output in the `arm64` build:

<img width="1263" alt="image" src="https://github.com/Automattic/studio/assets/1218433/a947c86b-19d4-4456-ab1b-4c2aa73913d0">

That is encouraging 🙂

Unfortunately, I don't have much to add in terms of addressing the other issues...

mokagio committed:
Conditionally rebuild `fs-xattr` before packaging DMG to fix dlopen issue (#321)

* Install Rosetta on macOS

Try to address
https://buildkite.com/automattic/studio/builds/1543#01904ffb-8498-4d13-aa5a-e2b3c6dbb302

* Rebuild fs-xatrr after binary has been built, conditionally

* Revert "Install Rosetta on macOS"

This reverts commit 8a4cab7724d52d24e9af1015aee5597d510e8f1c.

fluiddot committed:
Address typos in Buildkite pipeline

fluiddot committed:
Avoid verbose output when rebuilding `fs-xattr`

fluiddot commented:
Merging this PR as I've tested DMG files in https://github.com/Automattic/studio/pull/321.
